lightningd/plugin.c

```diff
@@ -9,6 +9,12 @@
 #include <lightningd/json.h>
 #include <unistd.h>
 
+#if defined(__APPLE__) && defined(__MACH__)
```

Comment:
Why not just add all these includes unconditionally?

Reply:
as far as I understood the issue this problem only occurred if macos is used. that is why I added them solely for macos